File: frontend/src/services/enrollmentService.js
import axios from 'axios';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

const enrollmentService = {
  async enrollStudent(studentId, courseId) {
    const response = await axios.post(`${API_URL}/enrollments/enroll`, null, {
      params: { studentId, courseId }
    });
    return response.data;
  },

  async getEnrollmentsByStudent(studentId) {
    const response = await axios.get(`${API_URL}/enrollments/student/${studentId}`);
    return response.data;
  },

  async getEnrollmentsByCourse(courseId) {
    const response = await axios.get(`${API_URL}/enrollments/course/${courseId}`);
    return response.data;
  },

  async updateGrade(enrollmentId, grade, score) {
    const response = await axios.put(`${API_URL}/enrollments/${enrollmentId}/grade`, null, {
      params: { grade, score }
    });
    return response.data;
  },

  async removeEnrollment(enrollmentId) {
    const response = await axios.delete(`${API_URL}/enrollments/${enrollmentId}`);
    return response.data;
  }
};

export default enrollmentService;

File: backend/src/main/java/com/university/cms/service/EnrollmentService.java
package com.university.cms.service;

import com.university.cms.entity.Enrollment;
import com.university.cms.entity.User;
import com.university.cms.entity.Course;
import com.university.cms.repository.EnrollmentRepository;
import com.university.cms.repository.UserRepository;
import com.university.cms.repository.CourseRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class EnrollmentService {
    
    @Autowired
    private EnrollmentRepository enrollmentRepository;
    
    @Autowired
    private UserRepository userRepository;
    
    @Autowired
    private CourseRepository courseRepository;
    
    public Enrollment enrollStudent(Long studentId, Long courseId) {
        if (enrollmentRepository.existsByStudentIdAndCourseId(studentId, courseId)) {
            throw new RuntimeException("Student is already enrolled in this course");
        }
        
        User student = userRepository.findById(studentId)
                .orElseThrow(() -> new RuntimeException("Student not found"));
        
        Course course = courseRepository.findById(courseId)
                .orElseThrow(() -> new RuntimeException("Course not found"));
        
        Enrollment enrollment = new Enrollment(student, course);
        return enrollmentRepository.save(enrollment);
    }
    
    public List<Enrollment> getEnrollmentsByStudent(Long studentId) {
        return enrollmentRepository.findByStudentId(studentId);
    }
    
    public List<Enrollment> getEnrollmentsByCourse(Long courseId) {
        return enrollmentRepository.findByCourseId(courseId);
    }
    
    public Enrollment updateGrade(Long enrollmentId, String grade, Double score) {
        Enrollment enrollment = enrollmentRepository.findById(enrollmentId)
                .orElseThrow(() -> new RuntimeException("Enrollment not found"));
        
        enrollment.setGrade(grade);
        enrollment.setScore(score);
        
        return enrollmentRepository.save(enrollment);
    }
    
    public void removeEnrollment(Long enrollmentId) {
        enrollmentRepository.deleteById(enrollmentId);
    }
    
    public Optional<Enrollment> getEnrollmentById(Long id) {
        return enrollmentRepository.findById(id);
    }
}
